internal/controller/pipelinerollout_controller.go

```diff
@@ -434,8 +385,3 @@ func applyPipelineSpec(
 	processPipelineStatus(ctx, pipeline, pipelineRollout)
```

Comment:
this is a separate issue outside of this PR, but I think we should move this outside of this function. Processing the pipeline status is something we should do independent of applying a Pipeline spec.

Reply:
any objections if I move that?

Reply:
none from me, it make sense